### PR TITLE
Improving the Link object with an easier to use API.

### PR DESCRIPTION
**Pull request by evert**

### CODE REVIEW

File: src/link.ts [hunk 1]
```diff
@@ -51,11 +51,17 @@ export type Link = {
 
 }
 
+type NewLink = Omit<Link, 'context'>;
```
Comment (simistern) [inline src/link.ts:54]: Could you describe what is happening here? A new link is omitting a Link type or context string?

File: test/unit/link.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  it('should instantiate', () => {
+
+    const links = new Links('http://base.example/');
+    expect(links).to.be.an.instanceof(Links);
+
+  });
+
+  it('should allow adding links', () => {
+
+    const links = new Links('http://base.example/');
```
Comment (simistern) [inline test/unit/link.ts:15]: As far as I understand this, are links now auto-instantiated with the base URL, and if a URL is not provided, it defaults to the base one of the original resource?

File: test/unit/link.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    });
+
+    expect(links.getAll()).to.eql([
+      {
+        context: 'http://base.example/',
+        rel: 'rel',
+        href: 'http://a.example/',
+      },
+      {
+        context: 'http://base.example/',
```
Comment (simistern) [inline test/unit/link.ts:29]: I may have missed this earlier, but what is the relationship between `context` and `href`?

File: src/link.ts (hunk 1 above)
Comment (evert) [reply src/link.ts:54]: We're creating a new type that removes the `context` property from `Link`. Below, we allow either `NewLink` or `Link`. The real effective goal here was to make `context` optional for callers.

File: test/unit/link.ts (hunk 2 above)
Comment (evert) [reply test/unit/link.ts:15]: Yes!

File: test/unit/link.ts (hunk 3 above)
Comment (evert) [reply test/unit/link.ts:29]: `context` is the base url basically.

File: src/link.ts (hunk 1 above)
Comment (simistern) [reply src/link.ts:54]: Interesting, what are the differences between this approach and simply rewriting Link to have the context property as optional?

Comment (evert) [reply src/link.ts:54]: If `context` is optional on the Link type it means anything that uses the Link type will now always need to check if context is defined or not. In reality context is only optional when adding it to the collection of links, any consumer of this object will guaranteed get a `context`.

**Pull request merged**